web/client/epics/search.js

```diff
@@ -44,21 +43,26 @@ const searchEpic = action$ =>
     .debounceTime(250)
     .switchMap( action =>
              // create a stream of streams from array
-             Rx.Observable.from((action.services || [ {type: "nominatim"} ])
-             // Create an stream for each Promise
-            .map( (service) => Rx.Observable.defer(() => services[service.type](action.searchText, service.options)
-                .then( (response= []) => response.map(result => ({...result, __SERVICE__: service, __PRIORITY__: service.priority || 0})) ))
-                .retryWhen(errors => errors.delay(200).scan((count, err) => {
-                    if ( count >= 2) {
-                        throw err;
-                    }
-                    return count + 1;
-                }, 0))
-            ))
+            Rx.Observable.from(
+                (action.services || [ {type: "nominatim"} ])
+                 // Create an stream for each Service
+                .map(
+                    (service) => Rx.Observable.defer(() => API.Utils.getService(service.type)(action.searchText, service.options)
+                    .then(
+                        (response= []) => response.map(result => ({...result, __SERVICE__: service, __PRIORITY__: service.priority || 0}))
+                    ))
+                    .retryWhen(errors => errors.delay(200).scan((count, err) => {
```

Comment:
indent back, retryWhen is applied to the Observable, not to the service's promise